virtualbox: 5.2.2 -> 5.2.4

flokli opened the pull request:
###### Motivation for this change
This bumps virtualbox to version 5.2.4, adds its license and me as a maintainer.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

flokli committed:
virtualbox: 5.2.2 -> 5.2.4

flokli committed:
virtualbox: add flokli as maintainer

flokli committed:
virtualbox: add license

orivej commented:
This package is especially sensitive to `nox-review`, and it has a NixOS test in `nixos/tests/virtualbox.nix`. Knowing that will help you as a maintainer :)

flokli committed:
nixos/tests/virtualbox: remove mknod /dev/vda1 call

With devtmpfs introduced in 0d27df280f7ed502bba65e2ea13469069f9b275a it is
created automatically.

See https://github.com/NixOS/nixpkgs/pull/32983#issuecomment-353703083

flokli commented:
@orivej Of course you're right, missed them :blush: Tests also showed I forgot to update the `sha256sum` of `virtualBoxGuestAdditions`, which is now fixed.

The tests seemed to `mknod /dev/vda1`, which already existed, so the tests failed (and most likely did before, too). This is fixed, too.

I have no idea why @GrahamcOfBorg complains about xen here, maybe @grahamc has an idea?

orivej commented:
Thank you! The tests were broken by 0d27df280f7ed502bba65e2ea13469069f9b275a (as can be seen at https://hydra.nixos.org/build/65928567). /cc @dezgeg (not to blame, but to inform).

orivej reviewed (approved):
I have fixed this up a bit and consider it ready to merge.

grahamc commented:
@GrahamcOfBorg eval

(sorry for the noise, master was broken by a merge)
